python/graphstorm/eval/evaluator.py

```diff
@@ -450,20 +462,272 @@ def compute_score(self, pred, labels):
+            early_stop = early_stop_avg_increase_judge(val_score,
+                self._val_perf_list, self.get_metric_comparator())
+        elif self._early_stop_strategy == EARLY_STOP_CONSECUTIVE_INCREASE_STRATEGY:
+            early_stop = early_stop_cons_increase_judge(val_score,
+                self._val_perf_list, self.get_metric_comparator())
+        else:
+            return False
+
+        self._val_perf_list.pop(0)
+        self._val_perf_list.append(val_score)
+
+        return early_stop
+
+    def get_metric_comparator(self):
+        """ Return the comparator of the major eval metric.
+
+            We treat the first metric in all evaluation metrics as the major metric.
+        """
+        assert self.metrics_obj is not None, \
+            "Evaluation metrics object should not be None"
+        metric = self.metric[0]
+        return self.metrics_obj.metric_comparator[metric]
+
+    def get_val_score_rank(self, val_score):
+        """
+        Get the rank of the given validation score by comparing its values to the existing value
+        list.
+
+        Parameters
+        ----------
+        val_score: dict
+            A dictionary whose key is the metric and the value is a score from evaluator's
+            validation computation.
+        """
+        val_score = list(val_score.values())[0]
+
+        rank = get_val_score_rank(val_score,
+                                  self._val_perf_rank_list,
+                                  self.get_metric_comparator())
+        # after compare, append the score into existing list
+        self._val_perf_rank_list.append(val_score)
+        return rank
+
+    @property
+    def metric(self):
+        """ evaluation metrics
+        """
+        return self._metric
+
+    @property
+    def best_val_score(self):
+        """ Best validation score
+        """
+        return self._best_val_score
+
+    @property
+    def best_test_score(self):
+        """ Best test score
+        """
+        return self._best_test_score
+
+    @property
+    def best_iter_num(self):
+        """ Best iteration number
+        """
+        return self._best_iter
+
+    @property
+    def history(self):
+        """ Evaluation history
+        
+            Returns
+            -------
+            A list of evaluation history in training. The detailed contents of the list rely
+            on specific Evaluators. For example, ``GSgnnRegressionEvaluator`` and
+            ``GSgnnAccEvaluator`` add a tuple of validation and testing score as one list element.
+        """
+        return self._history
+
+    @property
+    def eval_frequency(self):
+        """ Evaluation frequency
+        """
+        return self._eval_frequency
+
+    @property
+    def task_tracker(self):
+        """ Task tracker of this evaluator
+        """
+        return self.tracker
+
+class GSgnnRegressionEvaluator(GSgnnInstanceEvaluator):
```

Comment:
Should GSgnnRegressionEvaluator inherit GSgnnBaseEvaluator?

Reply:
And also GSgnnLPEvaluator?

Reply:
I will work on the Regression and LinkPrediction Evaluator in the next PR.